### PR TITLE
Add .clang-format and apply to dimod/include/*

### PR DESCRIPTION
**Pull request by arcondello**
See https://github.com/dwavesystems/dwave-ocean-sdk/pull/90 and https://github.com/dwavesystems/dwave-ocean-sdk/pull/103

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+Language: Cpp
+BasedOnStyle: Google
+
+ColumnLimit: 80
+NamespaceIndentation: Inner
+
+# Scaled by a factor of 2 such that the base indent is 4
+AccessModifierOffset: -3
```
Comment (randomir) [inline .clang-format:9]: Why not -2?

File: dimod/include/dimod/adjvectorbqm.h [hunk 2]
```diff
@@ -96,159 +98,156 @@ class AdjVectorBQM {
...
+    std::pair<outvars_iterator, outvars_iterator> neighborhood(
+        variable_type u) {
         assert(u >= 0 && u < adj.size());
         return std::make_pair(adj[u].first.begin(), adj[u].first.end());
     }
 
-    std::pair<const_outvars_iterator, const_outvars_iterator>
-    neighborhood(variable_type u) const {
+    std::pair<const_outvars_iterator, const_outvars_iterator> neighborhood(
+        variable_type u) const {
```
Comment (randomir) [inline dimod/include/dimod/adjvectorbqm.h:227]: This feels like it should be indented 8 spaces. Not sure is that due to `ContinuationIndentWidth` or `ConstructorInitializerIndentWidth` not scaled with `IndentWidth` in dwave-ocean-sdk#90.

File: .clang-format (hunk 1 above)
Comment (arcondello) [reply .clang-format:9]: Discussion: https://github.com/dwavesystems/dwave-ocean-sdk/pull/103#discussion_r513567538

**Comment by arcondello** [comment]
Ok, scaled the indent. @randomir , can you take another look? Figure it's better to iterate here than on dwavesystems/dwave-ocean-sdk#103

File: dimod/include/dimod/adjmapbqm.h [hunk 3]
```diff
@@ -95,108 +97,110 @@ class AdjMapBQM {
...
-      {
-        // Zero out the diagonal to avoid expensive checks inside innermost
-        // loop in the code for reading the matrix. The diagonal will be
-        // restored so a backup copy is saved.
-        #pragma omp for schedule(static)
-        for (size_type v = 0; v < num_variables; ++v) {
-          dense_diagonal[v] = dense[v * (num_variables + 1)];
-          dense[v * (num_variables + 1)] = 0;
+        if (!ignore_diagonal) {
+#pragma omp parallel for
```
Comment (arcondello) [inline dimod/include/dimod/adjmapbqm.h:107]: Annoys me that it cannot handle `#pragma omp parallel for` etc. There [are solutions](https://stackoverflow.com/questions/31353748/how-could-i-indent-c-pragma-using-clang-format) but probably not common enough to require.

**Pull request merged**